pytorch-0.4.1 fails to build

## Issue description

I cannot build pytorch (cuda enabled or disabled), due to failing tests. I'm not sure why this is built locally instead of taken from the cache.

### Steps to reproduce

`> nix run "(with import <nixpkgs> {}; python3.withPackages (ps: with ps; [ pytorch ]))"`

I expect to get a shell where `python3` can be run and `pytorch` can be imported. Instead, I get an error:

```
builder for '/nix/store/zj90cf6xa9p0m2lv701rhr5z7ddg8mwy-python3.6-pytorch-0.4.1.drv' failed with exit code 1; last 10 log lines:
  ----------------------------------------------------------------------
  Ran 41 tests in 482.056s
  
  FAILED (failures=32, skipped=9)
  Traceback (most recent call last):
    File "test/run_test.py", line 345, in <module>
      main()
    File "test/run_test.py", line 337, in main
      raise RuntimeError(message)
  RuntimeError: test_distributed failed!
```

Full log is too long for pastebin, but the errors mostly look like:

```
Process process 2:
Traceback (most recent call last):
  File "/nix/store/1c4zshbxizcsvyj46fwf5jj10h41wf6i-python3-3.6.6/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/nix/store/1c4zshbxizcsvyj46fwf5jj10h41wf6i-python3-3.6.6/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "test_distributed.py", line 950, in _run
    world_size=int(WORLD_SIZE))
  File "/nix/store/plrqrhi0npp7ba80chq1a49nhl5vnk5d-python3.6-pytorch-0.4.1/lib/python3.6/site-packages/torch/distributed/__init__.py", lin
e 94, in init_process_group
    group_name, rank)
RuntimeError: could not establish connection with other processes at /build/source/torch/lib/THD/process_group/General.cpp:17
F
======================================================================
FAIL: test_DistributedDataParallelCPU (__main__.TestDistBackend)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_distributed.py", line 911, in wrapper
    self._join_and_reduce(fn)
  File "test_distributed.py", line 971, in _join_and_reduce
    self.assertEqual(p.exitcode, first_process.exitcode)
  File "/build/source/test/common.py", line 364, in assertEqual
    super(TestCase, self).assertEqual(x, y, message)
AssertionError: 1 != None : 
```

With these probably relevant lines near the beginning:

```
Error: failed to parse the list of possible procesors in /sys/devices/system/cpu/possible
Error: failed to parse the list of present procesors in /sys/devices/system/cpu/present
```

Those two files both contain `0-11` in my user shell.

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.74, NixOS, 19.03pre154487.0a7e258012b (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(kai): `""`
 - channels(root): `"nixos-19.03pre154487.0a7e258012b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/pytorch/default.nix
+++ b/pkgs/development/python-modules/pytorch/default.nix
@@ -72,7 +72,7 @@ in buildPythonPackage rec {
   ] ++ lib.optional (pythonOlder "3.5") typing;
 
   checkPhase = ''
-    ${cudaStubEnv}python test/run_test.py --exclude dataloader sparse torch utils
+    ${cudaStubEnv}python test/run_test.py --exclude dataloader sparse torch utils distributed
   '';
 
   meta = {

